spec/lib/avalon/build_utils_spec.rb

```diff
@@ -0,0 +1,166 @@
+# Copyright 2011-2022, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+require 'avalon/build_utils'
+
+describe Avalon::BuildUtils do
+    before :all  do
+      Utils = Avalon::BuildUtils.new
+    end
```

Comment:
I think this would do well as a `subject`:
```suggestion
  subject(:utils) { described_class.new }
```
Then the tests would be like:
```
  expect(subject.detect_version(contents_arr)).to eq '1.2.3'
  # OR
  expect(utils.detect_version(contents_arr)).to eq '1.2.3'
```